lib/active_fedora/associations/has_and_belongs_to_many_association.rb

```diff
@@ -96,6 +82,12 @@ def stale_state
           owner[reflection.foreign_key]
         end
 
+        def find_target
+          ids = owner[reflection.foreign_key]
+          return [] if ids.blank?
+          ActiveFedora::Base.find(ids)
```

Comment:
@mjgiarlo It always was hitting Fedora. The call to `ActiveFedora::QueryResultBuilder.reify_solr_results` was internally issuing a `.find()` We are now avoiding the hit to Solr because `.find()` supports an array of ids.